src/services/createRoute.ts

```diff
@@ -16,6 +16,7 @@ import { Query, ToQuery, toQuery } from '@/types/query'
 import { RouteMeta } from '@/types/register'
 import { Route } from '@/types/route'
 import { checkDuplicateParams } from '@/utilities/checkDuplicateKeys'
+import { WithHooks } from '@/types/hooks'
```

Comment:
I don't really like that this type is moved and used in two places (here and in `RouterOptions`). Maybe this should remain here and just be duplicated in `RouterOptions`?